datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1192,7 +1192,11 @@ fn ensure_distribution(
     .collect::<Result<Vec<_>>>()?;
 
     let children_plans = children.iter().map(|c| c.plan.clone()).collect::<Vec<_>>();
-    plan = if plan.as_any().is::<UnionExec>() && can_interleave(children_plans.iter()) {
+
+    plan = if plan.as_any().is::<UnionExec>()
+        && !config.optimizer.prefer_existing_union
```

Comment:
Given that one of the motivations (https://github.com/influxdata/arrow-datafusion/pull/4) for this new flag is to preserve the sorting of the Union - would re-using the existing flag `prefer_existing_sort` make sense here?

One argument against that would be if you wanted `prefer_existing_sort: false` and `prefer_existing_union: true`.

Reply:
> Given that one of the motivations ([influxdata#4](https://github.com/influxdata/arrow-datafusion/pull/4)) for this new flag is to preserve the sorting of the Union - would re-using the existing flag `prefer_existing_sort` make sense here?
>
> One argument against that would be if you wanted `prefer_existing_sort: false` and `prefer_existing_union: true`.

This is an excellent point @phillipleblanc .  I just double checked and we actually have `prefer_existing_sort` set to true in IOx already ([code ref, not public](https://github.com/influxdata/influxdb_iox/blob/ec0b64557a70fc262ca09e6a6c941ae642667f31/datafusion_util/src/config.rs#L34))

What do you think about using the existing flag @NGA-TRAN ?

Reply:
@phillipleblanc and @alamb 
@mustafasrepo 's comment https://github.com/apache/datafusion/pull/10259#issuecomment-2082318558 suggest to use the current approach. Do I need to do anything more here?

Reply:
Nope, think this approach looks good! Thanks!